File: test-case/common/intbreak/Intbreak.java
import soda.unittest.*;

import java.util.*;
import java.util.function.Supplier;
import java.util.stream.*;

import soda.leetcode.*;
import soda.unittest.Validators;
import soda.unittest.TestWork;

class Solution {
    private int[] memo;
    public int integerBreak(int n) {
        memo = new int[59];
        return solve(n);
    }

    private int solve(int n) {
        if (n == 1) {
            return 1;
        }
        if (memo[n] > 0) {
            return memo[n];
        }
        int res = 0;
        for (int i = 1; i < n; ++i) {
            res = Math.max(i * (n-i), res);
            res = Math.max(i * solve(n-i), res);
        }
        memo[n] = res;
        return res;
    }
}

public class Intbreak implements Supplier<TestWork> {

    @Override
    public TestWork get() {
        var work = new TestWork(new Solution(), "integerBreak");
        // var work = TestWork.forStruct(Struct.class);
        // work.setValidator((e, r) -> {...});
        work.setCompareSerial(true);
        return work;
    }

    public static void main(String[] args) throws Exception {
        new Intbreak().get().run();
    }

}

File: test-case/common/intbreak/intbreak.go
package main

import (
    . "missingjs.com/soda/leetcode"
    "missingjs.com/soda/unittest"
    "missingjs.com/soda/util"
)

var logger = util.Logger
var _used ListNode

var memo []int

func integerBreak(n int) int {
    memo = make([]int, 59)
    return solve(n)
}

func solve(n int) int {
    if n == 1 {
        return 1
    }
    if memo[n] > 0 {
        return memo[n]
    }
    res := 0
    for i := 1; i < n; i++ {
        res = max(res, i * (n-i))
        res = max(res, i * solve(n-i))
    }
    memo[n] = res
    return res
}

func max(a, b int) int {
    if a > b {
        return a
    }
    return b
}

func main() {
    // create tester by function
    work := unittest.CreateWork(integerBreak)
    // OR create by struct factory
    // work := unittest.CreateWorkForStruct(Constructor)
    // work.SetValidator(func(e,r)bool)
    work.CompareSerial = true
    work.Run()
}
